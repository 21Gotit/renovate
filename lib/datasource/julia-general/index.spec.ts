import fs from 'fs';
import { DATASOURCE_FAILURE } from '../../constants/error-messages';
import _got from '../../util/got';
import { getReleases } from '.';

const got: jest.Mock<any> = _got as any;
jest.mock('../../util/got');

const res1 = fs.readFileSync(
  'lib/datasource/julia-general/__fixtures__/GaussianProcesses.toml',
  'utf8'
);

describe('datasource/julia-general', () => {
  describe('getReleases', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });
    it('returns null for 404', async () => {
      got.mockRejectedValueOnce({ statusCode: 404 });
      expect(await getReleases({ lookupName: 'foo/bar' })).toBeNull();
    });
    it('returns null for empty 200 OK', async () => {
      got.mockResolvedValueOnce({ body: {} });
      expect(
        await getReleases({ lookupName: 'doesnotexist/doesnotexist' })
      ).toBeNull();
    });
    it('throws for 429', async () => {
      got.mockRejectedValueOnce({ statusCode: 429 });
      await expect(getReleases({ lookupName: 'foo/bar' })).rejects.toThrow(
        DATASOURCE_FAILURE
      );
    });
    it('processes real data', async () => {
      got.mockResolvedValueOnce({ body: res1 });
      const res = await getReleases({ lookupName: 'GaussianProcesses' });
      expect(res).toMatchSnapshot();
    });
  });
});
