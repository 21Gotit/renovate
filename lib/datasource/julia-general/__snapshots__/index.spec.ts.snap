// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datasource/julia-general getReleases processes real data 1`] = `
Object {
  "releases": Array [
    Object {
      "newDigest": "7c7ff029888c9ce6a4db84cf11b11f8c4aaaa190",
      "version": "0.8.0",
    },
    Object {
      "newDigest": "d27b78ff418d683df83ea7ae74f4eb545a8504ae",
      "version": "0.9.0",
    },
    Object {
      "newDigest": "f9f6ab756930c66824e5b374186750e0fb722caf",
      "version": "0.10.0",
    },
    Object {
      "newDigest": "9d6810e16618447a4bd77687be35568ed45c95cf",
      "version": "0.11.0",
    },
    Object {
      "newDigest": "10fd495d82fb3fa0b0c1293e988e253870f2fa48",
      "version": "0.11.1",
    },
    Object {
      "newDigest": "105c668971f973b0f1b84931b0d68b94da78e692",
      "version": "0.11.2",
    },
    Object {
      "newDigest": "78a9285fecbeee4849023f1f1906db9fd765d768",
      "version": "0.12.0",
    },
    Object {
      "newDigest": "9b33cd8be0fd2bc59c46911def57331a7d5f6e84",
      "version": "0.12.1",
    },
  ],
}
`;
